File: android/src/main/java/com/davecoates/rnfirebasebridge/FirebaseBridgeTwitterAuthProvider.java
package com.davecoates.rnfirebasebridge;
import com.facebook.react.bridge.*;
import com.google.firebase.auth.AuthCredential;
import com.google.firebase.auth.TwitterAuthProvider;

/**
 * Created by dave on 20/08/16.
 */
public class FirebaseBridgeTwitterAuthProvider extends ReactContextBaseJavaModule {

    public FirebaseBridgeTwitterAuthProvider(ReactApplicationContext reactContext) {
        super(reactContext);
    }

    @Override
    public String getName() {
        return "FirebaseBridgeTwitterAuthProvider";
    }

    @ReactMethod
    public void credential(String token, String secret, Promise promise) {
        AuthCredential credential = TwitterAuthProvider.getCredential(token, secret);
        WritableMap data = Arguments.createMap();
        data.putString("id", FirebaseBridgeCredentialCache.addCredential(credential));
        data.putString("provider", credential.getProvider());
        promise.resolve(data);
    }
}

File: ios/FirebaseBridgeTwitterAuthProvider.swift
//
//  FirebaseBridgeAuth.swift
//
//  Created by Dave Coates on 20/08/2016.
//

import Firebase

@objc(FirebaseBridgeTwitterAuthProvider)
class FirebaseBridgeTwitterAuthProvider : NSObject {
  
  var bridge: RCTBridge!
  
  override init() {
    super.init()
  }
  
  @objc func credential(_ token:String, secret:String, resolver resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) {
    let credential = FIRTwitterAuthProvider.credential(withToken: token, secret: secret)
    resolve([
      "id": FirebaseBridgeCredentialCache.addCredential(credential),
      "provider": credential.provider,
      ])
  }
}





